### ISSUE TITLE
[Bug]: Yubikey Webauthn 2FA fails to authenticate.

### ISSUE THREAD

**Issue by 0x1a8510f2** [zitadel/zitadel#5971]
### Preflight Checklist

- [X] I could not find a solution in the documentation, the existing issues or discussions
- [ ] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Environment

Self-hosted

### Version

2.28.0

### Describe the problem caused by this bug

I had my Yubikey set up as a second factor for quite a while and never had any issues until now.

When attempting to authenticate as of late, touching my Yubikey produces the following error:

![image](https://github.com/zitadel/zitadel/assets/83190128/cac61970-da9e-4494-a971-44a31b5bc232)

I have confirmed that the Yubikey is working correctly by authenticating on another site.

I am also definitely not using the wrong key because I used the workaround below and then added the key once again which made no difference.

### To reproduce

I'm not entirely sure what may have caused this or how it can be reproduced. I am using the current latest version of Zitadel in Docker (`ghcr.io/zitadel/zitadel:latest`) with postgres 14 (I realise pg is experimental, but I highly doubt the issue is related).

The logs do not appear interesting but I attach them below anyway.

### Operating System

OS: openSUSE Tumbleweed Linux (latest snapshot)
Browser: Brave 1.52.117 (Chromium 114.0.5735.90)

### Additional Context

<details>
<summary>Logs</summary>
<pre><code>time="2023-06-03T22:33:56Z" level=info msg="setup started" caller="/home/runner/work/zitadel/zitadel/cmd/setup/setup.go:63"
time="2023-06-03T22:33:56Z" level=warning msg="postgres is currently in beta" caller="/home/runner/work/zitadel/zitadel/internal/database/postgres/config.go:62"
time="2023-06-03T22:33:56Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39" name=01_tables
time="2023-06-03T22:33:56Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39" name=02_assets
time="2023-06-03T22:33:56Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39" name=03_default_instance
time="2023-06-03T22:33:56Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39" name=04_eventstore_indexes
time="2023-06-03T22:33:56Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39" name=05_last_failed
time="2023-06-03T22:33:56Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39" name=06_resource_owner_columns
time="2023-06-03T22:33:56Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39" name=07_logstore
time="2023-06-03T22:33:56Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39" name=08_auth_token_indexes
time="2023-06-03T22:33:56Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39" name=09_optimise_indexes
time="2023-06-03T22:33:56Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39" name=10_correct_creation_date
time="2023-06-03T22:33:56Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39" name=11_event_created_at
time="2023-06-03T22:33:56Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39" name=config_change
time="2023-06-03T22:33:56Z" level=info msg="verify migration" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39" name=projection_tables
time="2023-06-03T22:33:56Z" level=warning msg="postgres is currently in beta" caller="/home/runner/work/zitadel/zitadel/internal/database/postgres/config.go:62"
time="2023-06-03T22:33:57Z" level=warning msg="execution logs are currently in beta" caller="/home/runner/work/zitadel/zitadel/cmd/start/start.go:207"
time="2023-06-03T22:33:57Z" level=info msg="spooler started" caller="/home/runner/work/zitadel/zitadel/internal/eventstore/v1/spooler/spooler.go:73" lockerID=216989193186844994 workers=1
time="2023-06-03T22:33:57Z" level=info msg="spooler started" caller="/home/runner/work/zitadel/zitadel/internal/eventstore/v1/spooler/spooler.go:73" lockerID=216989193186910530 workers=1
time="2023-06-03T22:33:58Z" level=info msg="server is listening on [::]:8080" caller="/home/runner/work/zitadel/zitadel/cmd/start/start.go:412"</code></pre>
</details>

### Workaround

The 2FA can temporarily be disabled by manually editing the database. This can be dangerous, but seems to have worked fine for me:

```
UPDATE auth.users2 SET u2f_tokens=null WHERE preferred_login_name='user@domain.com';
UPDATE auth.users SET u2f_tokens=null WHERE user_name='user@domain.com';
```

**Comment by hifabienne** [comment]
Hei @0x1a8510f2 
Thanks for reporting.
We will have a look at it.

**Comment by hifabienne** [comment]
@peintnermax @muhlemmer Can you link the pr that fixed that issue and close it if it is already done?

**Comment by peintnermax** [comment]
#5983 fixes the decoding of the credential id

**Comment by muhlemmer** [comment]
Workaround fix (revert) was already part of the 2.28.1 image. The final fix from Max will be part of version 2.29. 

Closing as completed.

**Comment by 0x1a8510f2** [comment]
Can confirm, the latest docker image appears to work as expected. Thanks for the fix!
